public/js/controllers/tx_controller.js

```diff
@@ -0,0 +1,92 @@
+import txInBlock from '../helpers/block_helper'
+import globalEventBus from '../services/event_bus_service'
+import { Controller } from 'stimulus'
+import humanize from '../helpers/humanize_helper'
+
+export default class extends Controller {
+  static get targets () {
+    return ['unconfirmed', 'confirmations', 'formattedAge', 'age', 'progressBar',
+      'ticketStage', 'poolStatus', 'expiryBar']
+  }
+
+  connect () {
+    this.advanceTicketProgress = this._advanceTicketProgress.bind(this)
```

Comment:
Why is .bind(this) being used on controller methods?
Methods declared on the controller class are already scoped to the controller, so it seems redundant to do this. 

Also, what is the leading underscore supposed to indicate?

Reply:
> Methods declared on the controller class are already scoped to the controller

This is not true when used as event callbacks.

```
myFunc () {
    // do stuff
}

myCallback (e) {
    // do stuff
    this.myFunc()
}

connect () {
    globalEventBus.on('BLOCK_RECEIVED', this.myCallback)
}
```
won't work (`Cannot read property 'myFunc' of undefined`).

Typically, I'd just define is as an arrow function in the argument

```
globalEventBus.on('BLOCK_RECEIVED', (e) => {
    // do stuff
})
```

but I can't here because I need to retain a reference for unbinding on disconnect.

I could assign an arrow function to a property in `connect`/`initialize`.

```
connect () {
    this.myFunc = (e) => {
        //do stuff or call class method
    }
    globalEventBus.on('BLOCK_RECEIVED', this.myFunc)
}

disconnect () {
    globalEventBus.off('BLOCK_RECEIVED', this.myFunc)
}
```

But I used bind because I think it's cleaner.

Reply:
Ah right, that makes perfect sense, I think you're taking the right approach then. Thanks for the explanation.